README.md

```diff
@@ -42,18 +63,90 @@ The Operator Acts on the following CRDs:
 
-__The Cluster state includes:__
-- OFED driver container to be deployed on Mellanox supporting nodes
-- GPUDirect driver container to be deployed on RDMA & GPU supported nodes
-- RDMA shared device plugin and related configurations
+#### NICClusterPolicy spec:
+NICClusterPolicy CRD Spec includes the following sub-states/stages:
+- `ofedDriver`: [OFED driver container](https://github.com/Mellanox/ofed-docker) to be deployed on Mellanox supporting nodes.
+- `devicePlugin`: [RDMA shared device plugin](https://github.com/Mellanox/k8s-rdma-shared-dev-plugin)
+and related configurations.
+- `nvPeerDriver`: [Nvidia Peer Memory client driver container](https://github.com/Mellanox/ofed-docker)
+to be deployed on RDMA & GPU supporting nodes (required for GPUDirect workloads).
+
+>__NOTE__: Any sub-state may be omitted if it is not required for the cluster.
+
+##### Example for NICClusterPolicy resource:
+In the example below we request OFED driver to be deployed together with RDMA shared device plugin
+but without NV Peer Memory driver.
+
+```
+apiVersion: mellanox.com/v1alpha1
+kind: NicClusterPolicy
+metadata:
+  name: example-nicclusterpolicy
+  namespace: mlnx-network-operator
+spec:
+  ofedDriver:
+    image: ofed-driver
+    repository: mellanox
+    version: 5.0-2.1.8.0
+  devicePlugin:
+    image: k8s-rdma-shared-dev-plugin
+    repository: mellanox
+    version: latest
+    # The config below directly propagates to k8s-rdma-shared-device-plugin configuration.
+    # Replace 'devices' with your (RDMA capable) netdevice name.
+    config: |
+      {
+        "configList": [
+          {
+            "resourceName": "hca_shared_devices_a",
+            "rdmaHcaMax": 1000,
+            "devices": ["ens2f0"]
+          }
+        ]
+      }
+```
+
+Can be found at: `example/deploy/crds/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml`
+
+#### NICClusterPolicy status:
+NICClusterPolicy `status` field reflects the current state of the system.
+it Contains a per sub-state/stage and a global state `status` that indicates if the cluster has transitioned to the desired
+state e.g OFED driver container deployed and loaded on relevant nodes, RDMA device plugin deployed
+and running on relevant nodes.
+The global state reflects the logical _AND_ of each inidvidual sub-state.
+
+##### Example of the `Status` field of a NICClusterPolicy instance:
+```
+Status:                                    
+  Applied States:
+    Name:   state-OFED
+    State:  ready
+    Name:   state-RDMA-device-plugin
+    State:  ready
+    Name:   state-NV-Peer
+    State:  ignore
+  State:    ready
+```
+
+>__NOTE__: An `ignore` State indicates that the sub-state/stage was not defined in the custom resource
+> this it is ignored.
 
 ## System Requirements
-As more driver containers are built the operator will be able to support additional platforms.
-for Phase 1 Ubuntu20.04 is planned to be supported.
+* RDMA capable hardware: Mellanox ConnectX-5 NIC.
```

Comment:
We can say it will work on Mellanox ConnectX-4 and above and just tested in ConnectX-5

Reply:
sure

Reply:
i dont think we need to say here on what we tested